daffodil-test/src/test/resources/org/apache/daffodil/section15/choice_groups/choice.tdml

```diff
@@ -1838,6 +1838,7 @@ it sure is/
     <tdml:errors>
       <tdml:error>Schema Definition Error</tdml:error>
       <tdml:error>choiceBranchKey</tdml:error>
+      <tdml:error>choiceBranchKeyRanges</tdml:error>
       <tdml:error>not defined</tdml:error>
     </tdml:errors>
   </tdml:parserTestCase>
```

Comment:
I didn't see a test for the range list being odd length, though your code does detect that and issue an SDE.